lib/models/authenticator.js

```diff
@@ -304,13 +310,27 @@ class TlsAuthenticator extends Authenticator {
    * @return {UserAccount}
    */
   ensureLocalUser (webId) {
-    if (this.accountManager.externalAccount(webId)) {
-      debug(`WebID URI ${JSON.stringify(webId)} is not a local account`)
+    const serverUri = this.accountManager.host.serverUri
 
-      throw new Error('Cannot login: Selected Web ID is not hosted on this server')
+    // if (this.accountManager.externalAccount(webId)) {
+    if (domainMatches(serverUri, webId)) {
+      // This is a locally hosted Web ID
+      return Promise.resolve(this.accountManager.userAccountFrom({ webId }))
     }
 
-    return this.accountManager.userAccountFrom({ webId })
+    debug(`WebID URI ${JSON.stringify(webId)} is not a local account, verifying preferred provider`)
+
+    return this.discoverProviderFor(webId)
+      .then(preferredProvider => {
```

Comment:
I wonder is "preferred" is too strong, given that we only have a predicate to indicate the provider (and we simply assume there is only one).